### DISCUSSION TITLE
Identifying colored page and updating density

### DISCUSSION THREAD

**Discussion by hammersxo** [ImageMagick/ImageMagick#2448] [General]
I am using Magick.NET-Q8-AnyCPU nuget package in .Net core application hosted on windows and also testing it on Linux docker container.
Requirement is to identify colored pages from a multipage tiff and update resolution only for that.
TIFF can have 800 pages and both B/W as well colored pages.
I am using the below code to load single page at a time and then check for its color.
For 800 pages, this method takes around 1min.
Is there anything that I can modify here so that the entire operation is faster.

    public void ResaveFileUsingMagickPageSettings(string tiffFile)
    {
        MagickImageCollection imageCollection = new MagickImageCollection();
        MagickImage imageTemp = new MagickImage();

        // MagickNET.SetTempDirectory(@"/app/TiffImages/Temp");

        var readSettings = new MagickReadSettings
        {
            FrameCount = 1,
            FrameIndex = 1
        };

        using (var tif = new MagickImageCollection(tiffFile, readSettings))
        {
            tif.Ping(tiffFile);
            for (int i = 0; i < tif.Count; i++)
            {
                MagickReadSettings settings = new MagickReadSettings()
                {
                    FrameIndex = i,
                    Density = new Density(50, 50),
                    FrameCount = 1,
                };
                settings.SetDefine(MagickFormat.Tif, "ignore-layers", true);

                try
                {
                    //MagickImage image = new MagickImage(tiffFile, settings);
                    using (MagickImage image = new MagickImage(tiffFile, settings))
                    {
                        if (image.ColorSpace == ImageMagick.ColorSpace.sRGB)
                        {
                            image.Settings.Compression = CompressionMethod.LZW;

                            Density density = new Density(100, 100);
                            image.Density = density;
                            image.Resize(850, 1100);
                            image.ColorType = ColorType.Palette;
                            image.Depth = 8;
                        }

                        imageTemp = (MagickImage)image.Clone();
                        imageCollection.Add(imageTemp);
                    }
                }
                catch (MagickException ex)
                {

                }
            }
            imageCollection.Write(@"/app/TiffImages/Temp/test.tif");
            imageTemp.Dispose();
        }
    }

**Comment by snibgo** [comment]
As you don't need to read the pixels to get the colorspace, perhaps you can "ping" each image, then read it only if you need to.
